### ISSUE TITLE
Add support for getting and replacing topics on a repository

### ISSUE THREAD

**Issue by peterkline** [PyGithub/PyGithub#634]
Topics are a fairly new feature within Github.  Currently this functionality is provided in the API in a beta mode.  We have a need to use these to add attributes and then consume them for automation.  Hopefully this helps others.

API Documentation:
[List topics in a repo](https://developer.github.com/v3/repos/#list-all-topics-for-a-repository)
[Replace topics in a repo](https://developer.github.com/v3/repos/#replace-all-topics-for-a-repository)

Examples of use:
```python
# Get list of topics from a repository    
topics = repo_obj.get_topics()

# Replace current (if there are any) topics with list
repo_obj.replace_topics(['base', 'personal', 'playground'])
```

**Comment by TheVetter** [comment]
Has topics support been added?

**Comment by sfdye** [comment]
@TheVetter Should be soon, I have reviewed the PR and still waiting response for the original author @peterkline

**Comment by migueleliasweb** [comment]
While this issue is not fixed. we should be able to access the topics with the `_rawData`, right ?

**Comment by sfdye** [comment]
@migueleliasweb I am afraid not, as it requires calling another endpoint. (`/topics`) However, I will try to get this merged soon.

**Comment by sfdye** [comment]
@migueleliasweb You are right, the `topics` attributes is currently in preview, but to edit the attributes we still need to go back to `/topics` endpoint.

**Comment by migueleliasweb** [comment]
@sfdye I just wanted to list them =/

Anyway... I seem unable to get them even accessing the `_rawData` directly. See:

```python
ipdb> ll
    643     def get_repo(self, name):
    644         """
    645         :calls: `GET /repos/:owner/:repo <http://developer.github.com/v3/repos>`_
    646         :param name: string
    647         :rtype: :class:`github.Repository.Repository`
    648         """
    649         assert isinstance(name, str), name
    650         headers, data = self._requester.requestJsonAndCheck(
    651             "GET",
    652             "/repos/" + self.login + "/" + name
    653         )
--> 654         return github.Repository.Repository(self._requester, headers, data, completed=True)
    655 

ipdb> data["topics"]
*** KeyError: 'topics'
```

**Comment by sfdye** [comment]
@migueleliasweb You're missing the custom header `application/vnd.github.mercy-preview+json`

**Comment by migueleliasweb** [comment]
@sfdye Pardon me, I totally missed that !

**Comment by edanhauon** [comment]
Hi @sfdye 
How soon is soon?

I ask since My app is dependent on this matter - and if it won't happen soon we'll have to develop this ourselves.

Your help is much appreciated 👍.

Edan

**Comment by sfdye** [comment]
@edanhauon The original author is not responding, so if you could open a new PR I am happy to review and cut a new version once that's merged.

**Comment by edanhauon** [comment]
Could you maybe add me to the current PR so I could what has been done already?

Thanks,
Edan

**Comment by sfdye** [comment]
@edanhauon what do you mean?

**Comment by edanhauon** [comment]
@sfdye never mind, I see it in this thread. thanks.

**Comment by peterkline** [comment]
Sorry guys for not being around.  Been buried with work related coding.  if you can't wait for about 12 hours for me to make any mods and redo the PR then by all means fork off whats here.  If you can wait I will finish the coding.

**Comment by sfdye** [comment]
@peterkline I can wait

**Comment by wyardley** [comment]
Patch works for me, with the notes mentioned in the PR, that is, it would be a little more user-friendly if you can pass that array back rather than having to send it as `{'name': topics}`, so like so:
```
topics = repo.get_topics()
topics.append('foobar')
repo.replace_topics(data)
```
vs
```
topics = repo.get_topics()
topics.append('foobar')
data = {'names': topics}
repo.replace_topics(data)
```

**Comment by sfdye** [comment]
@wyardley suggestion accepted

**Comment by sfdye** [comment]
Anyone care to get this thing moving? Topic is a stable feature now, I wanna make it happen!

**Comment by wyardley** [comment]
I'm proposing #832, which resolves the test failure, and I think resolves most of this feedback? It also switches the input to `replace_topics()` to the expected array as discussed. Would appreciate feedback, esp. on the test case I added.

**Comment by sfdye** [comment]
Merged in #832 thanks to @wyardley and credits to @peterkline for the original PR 🎉
